frontend-opal/src/components/PortfolioContainer/AllocationInfo/AllocationInfo.js

```diff
@@ -1,60 +1,106 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 import { connect } from 'react-redux';
-import { View, Text, TouchableOpacity } from 'react-native';
-import { Card } from 'native-base';
+import { View, Text } from 'react-native';
+import { Card, Button } from 'native-base';
+import palette from 'google-palette';
+import getTotalInvestment from 'components/common/getTotalInvestment';
 import styles from './styles';
 
 import PieChart from '../PieChart';
 
-const AllocationInfo = ({ pieData, pieColor, legendData }) => (
-  <View style={styles.allocationArea}>
-    <Text style={styles.title}>Allocation</Text>
-
-    <Card>
-      <View style={styles.pie}>
-        <PieChart data={pieData} color={pieColor} legendData={legendData} />
-      </View>
-
-      <View style={styles.allocationButtons}>
-        <TouchableOpacity>
-          <Text style={styles.button}>SECTOR</Text>
-        </TouchableOpacity>
-        <TouchableOpacity>
-          <Text style={styles.button}>GEOGRAPHY</Text>
-        </TouchableOpacity>
-      </View>
-    </Card>
-  </View>
-);
+const AllocationInfo = ({ fetchPortfolio }) => {
+  let sectors = {};
+  let totalInvestment = '';
+  if (fetchPortfolio[0]) {
+    totalInvestment = getTotalInvestment(fetchPortfolio);
+  }
+
+  if (fetchPortfolio[0]) {
+    sectors = fetchPortfolio.reduce((sectorsList, instrument) => {
+      const result = sectorsList;
+      if (result[instrument.sector]) {
+        result[instrument.sector].list.push(instrument);
+      } else {
+        result[instrument.sector] = { list: [instrument] };
+      }
+      return result;
+    }, {});
+  }
+
+  const legendData = Object.keys(sectors).map(sector => ({
+    name: sector,
+    symbol: { type: 'square' }
+  }));
+
+  if (Object.keys(sectors).length !== 0) {
+    Object.keys(sectors).forEach(sector => {
+      sectors[sector].value =
+        (
+          sectors[sector].list.reduce((accumulativeValue, instrument) => {
+            let result = accumulativeValue;
+            result += instrument.amount * instrument.marketValue;
+            return result;
+          }, 0) / totalInvestment
+        ).toFixed(2) * 100;
+    });
+  }
+
+  const pieData = Object.keys(sectors).map(sector => ({
+    x: sector,
+    y: sectors[sector].value,
+    label: `${sectors[sector].value}%`
+  }));
+
+  let pieColor = palette('tol-sq', Object.keys(sectors).length + 2);
+  pieColor = pieColor.slice(2).map(color => `#${color}`);
```

Comment:
@WenyuD instead of having all these calculations here, separate those into a container component, so this component actually gets the things needed to be displayed as props: `pieData, legendData, pieColors`

Reply:
something like this https://redux.js.org/basics/example#containers-visibletodolistjs